app/src/main/java/de/westnordost/streetcomplete/quests/way_light/AddWayLightFragment.java

```diff
@@ -0,0 +1,76 @@
+package de.westnordost.streetcomplete.quests.way_light;
+
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import java.util.List;
+
+import de.westnordost.osmapi.map.data.OsmElement;
+import de.westnordost.streetcomplete.R;
+import de.westnordost.streetcomplete.quests.YesNoQuestAnswerFragment;
+
+public class AddWayLightFragment extends YesNoQuestAnswerFragment
+{
+	public static final String OTHER_ANSWER = "OTHER_ANSWER";
+
+	@Override
+	public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState)
+	{
+		View view = super.onCreateView(inflater, container, savedInstanceState);
+		setTitle();
+		return view;
+	}
+
+	private void setTitle()
+	{
+		String name = getName(getOsmElement());
+		if (name != null && !name.trim().isEmpty())
+		{
+			setTitle(R.string.quest_way_light_named_title, name);
+		} else
+		{
+			setTitle(R.string.quest_way_light_title);
+		}
+	}
+
+	private String getName(OsmElement element)
+	{
+		return element != null && element.getTags() != null ? element.getTags().get("name") : null;
```

Comment:
This construct is used so often in different classes along with the name != null && trimmed not is empty check, I think it is time to put this code  into a static function that returns either the trimmed name or null if it is empty or null. Would you do that (and use the function in other quest forms that use this as well)?

The function could be either put into AbstractQuestAnswerFragment or a new *Util class.